### PR TITLE
fsharpLint: enable noPartialFunctions rule

### PR DESCRIPTION
**Pull request by aarani**

### CODE REVIEW

**Comment by knocte** [comment]
Please explain in commit message body why the FSharpx.Collection had to be upgraded, because for the occassional diff-reader dev it's not immediately obvious. After that, I merge; thanks!

File: NOnion/Network/TorCircuit.fs [hunk 1]
```diff
@@ -92,14 +98,20 @@ type TorCircuit
...
+                        | Some destinationNode ->
+                            reversedNodesList, destinationNode
                     | Some destination ->
                         nodesStates
-                        |> List.takeWhile(fun node -> node <> destination)
-                        |> List.append(List.singleton destination)
-                        |> List.rev,
+                        |> Seq.takeWhile(fun node -> node <> destination)
+                        |> Seq.append(Seq.singleton destination)
+                        |> Seq.rev,
```
Comment (knocte) [inline NOnion/Network/TorCircuit.fs:114]: there's no need to change this to Seq; you can still use Seq's functions on a List

File: NOnion/Network/TorCircuit.fs [hunk 2]
```diff
@@ -125,13 +137,13 @@ type TorCircuit
                     }
 
                 let rec encryptMessage
-                    (nodes: List<TorCircuitNode>)
+                    (nodes: seq<TorCircuitNode>)
                     (message: array<byte>)
                     =
-                    match List.tryHead nodes with
-                    | Some node ->
+                    match Seq.tryHeadTail nodes with
```
Comment (knocte) [inline NOnion/Network/TorCircuit.fs:143]: same here

File: NOnion/Network/TorCircuit.fs (hunk 1 above)
Comment (knocte) [reply NOnion/Network/TorCircuit.fs:114]: ah, fair enough

**Comment by aarani** [comment]
> Please explain in commit message body why the FSharpx.Collection had to be upgraded, because for the occassional diff-reader dev it's not immediately obvious. After that, I merge; thanks!

added.

**Pull request merged**